src/js/modules/articles/components/ArticlePage.js

```diff
@@ -56,25 +95,30 @@ const styles = {
   },
 };
 
-const ArticlePage = ({
-  classes,
-  article,
-  section,
-  sections,
-  media,
-  openSubscriptionModal,
-}) => {
-  if (!article) {
-    return <NotFoundPage />;
+const ArticlePage = ({ classes, data }) => {
+  if (data.loading) {
+    return null;
   }
+  data = humps.camelizeKeys(data);
+  const { articleBySlug } = data;
+  const { section } = articleBySlug;
+
+  const referencedArticleId = SPEC_REFERENCE_PATTERN.test(articleBySlug.content)
```

Comment:
Ugghhh. Stop using regexes when you should just correct the model. You're trying to fix a problem in the wrong domain. Open up your mind